### PR TITLE
Change tests for MS.AAD.5.4v1 for teams group consent to not-implemented as setting is deprecated

### LINKED ISSUE #1139: MS.AAD.5.4v1 setting discontinued by Microsoft
# 🐛 Summary #

What's wrong? Please be specific.
MS.AAD.5.4v1 notes Group owners SHALL NOT be allowed to consent to applications. An FCEB agency during a technical exchange meeting has noted that this option is no longer supported in the admin center and thus the policy needs to be updated to reflect the change. 

This includes removal of the current policy and checks and investigation into the new "Team owner consent settings" 

## To reproduce ##

1. Run Scubagear and receive a "FAIL" for MS.AAD.5.4v1 
2. Return to the admin center and view the notice that "Group owener consent settings have been removed and replaced with team owner consent settings". Screenshot attached below

image
image
image

## Expected behavior ##

What did you expect to happen that didn't?

PASS / FAIL test results based upon the previously set setting.

### PR DESCRIPTION
**Pull request by rmoffitt-m**
#  #
  

## 🗣 Description ##

Update rego for MS.AAD.5.4v1 to not-implemented as the group consent setting is discontinued.
In case the setting is not discontinued in other Azure implementations (e.g. GCC), the rule will continue to function as expected.

## 💭 Motivation and context ##

Changes the tests to not-implemented to avoid falsely failing a test that can no longer be performed. The baseline can be removed in a future release.
Closes #1139

## 🧪 Testing ##

Run AAD tests and verify that the test result for MS.AAD.5.4v1 is N/A instead of failed and criticality is not-implemented. Ensure tests can still successfully run.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] Unit tests added/updated to cover PowerShell and Rego changes.
- [x] Functional tests added/updated to cover PowerShell and Rego changes.
- [x] All relevant functional tests passed.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [x] PR passed smoke test check.
- [x] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [x] Resolved all merge conflicts on branch
- [x] Notified merge coordinator that PR is ready for merge via comment mention
- [x] Demonstrate changes to the team for questions and comments. 
    (Note: Only required for issues of size `Medium` or larger)

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: Testing/Functional/Products/TestPlans/aad.testplan.yaml [hunk 1]
```diff
@@ -689,36 +689,8 @@ TestPlan:
   - PolicyId: MS.AAD.5.4v1
     TestDriver: ScubaCached
```
[the remark this answers is not in the document]
Comment (rmoffitt-m) [reply Testing/Functional/Products/TestPlans/aad.testplan.yaml]: Thanks - changed accordingly

File: Testing/Functional/Products/TestPlans/aad.testplan.yaml [hunk 2]
```diff
@@ -689,36 +689,8 @@ TestPlan:
...
-                  - Name: BlockUserConsentForRiskyApps
-                    Value: "false"
-                  - Name: EnableAdminConsentRequests
-                    Value: "false"
-                  - Name: ConstrainGroupSpecificConsentToMembersOfGroupId
-                    Value: ""
+      - TestDescription: MS.AAD.5.4v1
+        Preconditions: []
         Postconditions: []
         ExpectedResult: true
```
[the remark this answers is not in the document]
Comment (rmoffitt-m) [reply Testing/Functional/Products/TestPlans/aad.testplan.yaml]: Corrections made in a new commit.

**Review by ahuynhMITRE — APPROVED** [review]
new results consistent with other not implemented policies. Create a new issue for the removal of the setting from the SCB set for the "m" release milestone with a reference to this PR and issue!

**Comment by tkol2022** [comment]
Nice work Rich. Seems to match up with other deprecations we have incorporated before.

**Comment by rmoffitt-m** [comment]
Created new issue to update baseline to wrap up todos for this issue. No rebase required for successful merge.

**Comment by rmoffitt-m** [comment]
rebased and all checks passed. ready for merge @nanda-katikaneni

**Pull request merged**